docs/sources/flow/tutorials/flow-by-example/processing-logs/index.md

```diff
@@ -0,0 +1,382 @@
+
+### Stage 5
+
+```river
+stage.labels {
+    values = {
+        level = "",
+    }
+}
+```
+
+This stage adds a label to the log using the same shorthand as above (so this is equivalent to using `values = { level = "level" }`). This stage adds a label with key `level` and the value of `level` in the extracted map to the log (`"info"` from our example log line). This stage does not modify the extracted map.
+
+### Stage 6
+
+```river
+stage.output {
+    source = "log_line"
+}
+```
+
+This stage uses the `log_line` value in the extracted map to set the actual log line that is forwarded to Loki. Rather than sending the entire JSON blob to Loki, we are only sending `original_log_line["log"]["message"]`, along with some labels that we attached.
+
+This stage does not modify the extracted map.
+
+## Putting it all together
+
+Now that we have all of the pieces, let's run the {{< param "PRODUCT_ROOT_NAME" >}} and send some logs to it. Modify `config.river` with the config from the previous example and start the {{< param "PRODUCT_ROOT_NAME" >}} with:
+
+```bash
+/path/to/agent run config.river
+```
+
+To get the current time in `RFC3339` format, you can run:
+
+```bash
+date -u +"%Y-%m-%dT%H:%M:%SZ"
+```
+
+Try executing the following, replacing the `"timestamp"` value:
+
+```bash
+curl localhost:9999/loki/api/v1/raw -XPOST -H "Content-Type: application/json" -d '{"log": {"is_secret": "false", "level": "debug", "message": "This is a debug message!"}, "timestamp": <YOUR TIMESTAMP HERE>}'
+```
+
+Now that we have sent some logs, let's see how they look in Grafana. Navigate to [localhost:3000/explore](http://localhost:3000/explore) and switch the Datasource to `Loki`. Try querying for `{source="demo-api"}` and see if you can find the logs you sent!
+
+Try playing around with the values of `"level"`, `"message"`, `"timestamp"`, and `"is_secret"` and see how the logs change. You can also try adding more stages to the `loki.process` component to extract more values from the logs, or add more labels.
+
+![Example Loki Logs](/media/docs/agent/screenshot-flow-by-example-processed-log-lines.png)
+
+## Exercise
+
+Since we are already using docker and docker exports logs, let's get those logs into Loki. You may want to refer to the [discovery.docker](https://grafana.com/docs/agent/<AGENT_VERSION>/flow/reference/components/discovery.docker/) and [loki.source.docker](https://grafana.com/docs/agent/<AGENT_VERSION>/flow/reference/components/loki.source.docker/) documentation. 
```

Comment:
```suggestion
Since you are already using Docker and Docker exports logs, let's get those logs into Loki.
You can refer to the [discovery.docker](https://grafana.com/docs/agent/<AGENT_VERSION>/flow/reference/components/discovery.docker/) and [loki.source.docker](https://grafana.com/docs/agent/<AGENT_VERSION>/flow/reference/components/loki.source.docker/) documentation for more information. 
```